active/radiobuttons/RadioButtons_Spec.md

```diff
@@ -169,84 +169,120 @@ If you find that you need wrapping behavior, perhaps a RadioButtons group isn't
 
 # API Details
 
-## RadioButtons
+## RadioButtons IDL
 
-```
-namespace Microsoft.UI.Xaml.Controls
+``` c++
+namespace MU_XC_NAMESPACE
 {
+
+[WUXC_VERSION_PREVIEW]
 [webhosthidden]
 [contentproperty("Items")]
+[MUX_PROPERTY_CHANGED_CALLBACK(TRUE)]
+[MUX_PROPERTY_CHANGED_CALLBACK_METHODNAME("OnPropertyChanged")]
 unsealed runtimeclass RadioButtons : Windows.UI.Xaml.Controls.Control
 {
     RadioButtons();
 
     Object ItemsSource;
     Windows.Foundation.Collections.IVector<Object> Items{ get; };
-    Windows.UI.Xaml.DataTemplate ItemTemplate;
+    [MUX_DEFAULT_VALUE("winrt::RadioButtonsElementFactory{}")]
+    Object ItemTemplate;
 
-    Windows.UI.Xaml.DependencyObject ContainerFromItem(Object item);
-    Windows.UI.Xaml.DependencyObject ContainerFromIndex(Int32 index);
+    Windows.UI.Xaml.UIElement ContainerFromIndex(Int32 index);
 
+    [MUX_DEFAULT_VALUE("-1")]
     Int32 SelectedIndex;
-    Object SelectedItem;
+    Object SelectedItem{ get; };
     event Windows.UI.Xaml.Controls.SelectionChangedEventHandler SelectionChanged;
 
-    Int32 MaxColumns;
+    [MUX_DEFAULT_VALUE("1")]
+    Int32 MaximumColumns;
     Object Header;
 
     static Windows.UI.Xaml.DependencyProperty ItemsSourceProperty{ get; };
     static Windows.UI.Xaml.DependencyProperty ItemsProperty{ get; };
     static Windows.UI.Xaml.DependencyProperty ItemTemplateProperty{ get; };
     static Windows.UI.Xaml.DependencyProperty SelectedIndexProperty{ get; };
     static Windows.UI.Xaml.DependencyProperty SelectedItemProperty{ get; };
-    static Windows.UI.Xaml.DependencyProperty MaxColumnsProperty{ get; };
+    static Windows.UI.Xaml.DependencyProperty MaximumColumnsProperty{ get; };
     static Windows.UI.Xaml.DependencyProperty HeaderProperty{ get; };
 }
-}
-```
-
-## RadioButtons Automation Peer
 
-```
-namespace Microsoft.UI.Xaml.Automation.Peers
-{
-[webhosthidden]
-unsealed runtimeclass RadioButtonsListViewItemAutomationPeer : Windows.UI.Xaml.Automation.Peers.ListViewItemAutomationPeer
-{
-    RadioButtonsListViewItemAutomationPeer(RadioButtonsListViewItem owner);
-}
 }
+
 ```
 
 ## RadioButtons Primitives
 
-```
-namespace Microsoft.UI.Xaml.Controls.Primitives
+``` c++
+namespace MU_XCP_NAMESPACE
 {
+[WUXC_VERSION_PREVIEW]
 [webhosthidden]
-unsealed runtimeclass RadioButtonsListView : Windows.UI.Xaml.Controls.ListView
+[default_interface]
+unsealed runtimeclass RadioButtonsElementFactory : MU_XC_NAMESPACE.ElementFactory
```

Comment:
@StephenLPeters does this type need to be public? If so, why? What do people do with it?

Reply:
It is the default value for the ItemTemplate property on RadioButtons. I think we could get away with moving this to the private namespace.

Reply:
But would it still be in the public API? You couldn't use it from XAML then, but RadioBUttons should be able to set it from its code-behind.

Reply:
I think it needs to be a winrt type to be assignable to the dependency property?